x-pack/plugins/observability/public/hooks/use_alert_index_names.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useFetcher } from './use_fetcher';
+import { callObservabilityApi } from '../services/call_observability_api';
+
+const NO_INDEX_NAMES: string[] = [];
+
+export function useAlertIndexNames() {
+  const { data: indexNames = NO_INDEX_NAMES } = useFetcher(({ signal }) => {
+    return callObservabilityApi('GET /api/observability/rules/alerts/dynamic_index_pattern', {
```

Comment:
Is it ok to hardcode this endpoint URL? Usually, this kind of code is shared centrally.

Reply:
Do you mean to pass it as a parameter of the hook? I think in this case the URL is directly tied to the hook because, as far as I understand, this is the endpoint that will return us the index names, so I don't think it will make sense to use the hook with a different URL.
What do you think @mgiota?

Reply:
@estermv I think it is fine to keep it here